src/config/compiler.rs

```diff
@@ -1,9 +1,45 @@
-use indexmap::IndexSet;
+// Workaround for a false positive. The function `create_data_dir_lock` cannot be const.
+#![allow(clippy::missing_const_for_fn)]
```

Comment:
You can also apply `allow` directives directly on the function in question to avoid the overly-broad allowance here.

Reply:
Chatted offline about this. Apparently it doesn't work when applied directly to the function. 🤔